cf/commands/domain/create_shared_domain.go

```diff
@@ -22,42 +25,83 @@ func init() {
 }
 
 func (cmd *CreateSharedDomain) MetaData() command_registry.CommandMetadata {
+	fs := make(map[string]flags.FlagSet)
+	fs["router-group"] = &cliFlags.StringFlag{Name: "router-group", Usage: T("Routes for this domain will be configured only on the specified router group")}
 	return command_registry.CommandMetadata{
 		Name:        "create-shared-domain",
 		Description: T("Create a domain that can be used by all orgs (admin-only)"),
-		Usage:       T("CF_NAME create-shared-domain DOMAIN"),
+		Usage:       T("CF_NAME create-shared-domain DOMAIN") + T(" [--router-group ROUTER_GROUP]"),
+		Flags:       fs,
 	}
 }
 
-func (cmd *CreateSharedDomain) Requirements(requirementsFactory requirements.Factory, fc flags.FlagContext) (reqs []requirements.Requirement, err error) {
+func (cmd *CreateSharedDomain) Requirements(requirementsFactory requirements.Factory, fc flags.FlagContext) ([]requirements.Requirement, error) {
 	if len(fc.Args()) != 1 {
-		cmd.ui.Failed(T("Incorrect Usage. Requires an argument\n\n") + command_registry.Commands.CommandUsage("create-shared-domain"))
+		cmd.ui.Failed(T("Incorrect Usage. Requires DOMAIN as an argument\n\n") + command_registry.Commands.CommandUsage("create-shared-domain"))
 	}
 
-	reqs = []requirements.Requirement{
-		requirementsFactory.NewLoginRequirement(),
+	requiredVersion, err := semver.Make("2.36.0")
+	if err != nil {
+		panic(err.Error())
 	}
-	return
+
+	var reqs []requirements.Requirement
+
+	if fc.String("router-group") != "" {
+		reqs = append(reqs, []requirements.Requirement{
+			requirementsFactory.NewMinAPIVersionRequirement("Option '--router-group'", requiredVersion),
+			requirementsFactory.NewLoginRequirement(),
+			requirementsFactory.NewRoutingAPIRequirement(),
+		}...)
+	} else {
+		reqs = append(reqs, requirementsFactory.NewLoginRequirement())
+	}
+
+	return reqs, nil
 }
 
 func (cmd *CreateSharedDomain) SetDependency(deps command_registry.Dependency, pluginCall bool) command_registry.Command {
 	cmd.ui = deps.Ui
 	cmd.config = deps.Config
 	cmd.domainRepo = deps.RepoLocator.GetDomainRepository()
+	cmd.routingApiRepo = deps.RepoLocator.GetRoutingApiRepository()
 	return cmd
 }
 
 func (cmd *CreateSharedDomain) Execute(c flags.FlagContext) {
+	var routerGroup models.RouterGroup
 	domainName := c.Args()[0]
+	routerGroupName := c.String("router-group")
+
+	if routerGroupName != "" {
+		err := cmd.routingApiRepo.ListRouterGroups(func(group models.RouterGroup) bool {
+			if group.Name == routerGroupName {
+				routerGroup = group
+				return false
+			}
+
+			return true
+		})
+
+		if err != nil {
+			cmd.ui.Failed(err.Error())
+		}
+
+		if routerGroup.Guid == "" {
```

Comment:
This is surprising; it relies on the fact that ListRouterGroups above did not find a matching router group. I would add a boolean of `found` or something to indicate that it wasn't found rather than relying on the default value.